onnxruntime/core/optimizer/conv_activation_fusion.cc

```diff
@@ -45,10 +31,15 @@ Status ConvActivationFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
     }
 
     const auto& next_node = *(node->OutputNodesBegin());
+
     if (next_node.GetExecutionProviderType() != node->GetExecutionProviderType()) {
       continue;
     }
 
+    if (!graph.GetNodeOutputsInGraphOutputs(*node).empty()) {
```

Comment:
Feels a bit not clear when we are calling this and when it's not needed (because it is covered by subsequent checks embedded in other functions).

Reply:
Not following. We check this upfront before doing all the work.

Reply:
I meant in the various transformers, sometimes we call it directly and others through other checks. Not important for the purpose of this PR.